sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala

```diff
@@ -372,6 +372,48 @@ class OrcQuerySuite extends QueryTest with BeforeAndAfterAll with OrcTest {
     }
   }
 
+  test("SPARK-16948. Check empty orc partitioned tables in ORC") {
+    withSQLConf((HiveUtils.CONVERT_METASTORE_ORC.key, "true")) {
+      withTempPath { dir =>
+        withTable("empty_orc_partitioned") {
+          spark.sql(
+            s"""CREATE TABLE empty_orc_partitioned(key INT, value STRING)
+                | PARTITIONED BY (p INT) STORED AS ORC
+              """.stripMargin)
+
+          val emptyDF = Seq.empty[(Int, String)].toDF("key", "value").coalesce(1)
+          emptyDF.createOrReplaceTempView("empty")
+
+          // Query empty table
+          val df = spark.sql(
+            s"""SELECT key, value FROM empty_orc_partitioned
+                | WHERE key > 10
+                      """.stripMargin)
+          checkAnswer(df, emptyDF)
+        }
+      }
+
+      withTempPath { dir =>
+        withTable("empty_text_partitioned") {
+          spark.sql(
+            s"""CREATE TABLE empty_text_partitioned(key INT, value STRING)
+                | PARTITIONED BY (p INT) STORED AS TEXTFILE
```

Comment:
Testing the textfile format sounds useless. We do not convert it to `LogicalRelation`.